src/Symfony/Component/Mailer/Bridge/Sendgrid/Transport/SendgridApiTransport.php

```diff
@@ -125,7 +126,13 @@ private function getPayload(Email $email, Envelope $envelope): array
                 continue;
             }
 
-            if ($header instanceof TagHeader) {
+            if ($name === 'send-at') {
+                if (!$header instanceof DateHeader) {
+                    throw new TransportException(sprintf('The Send-At header should be a "%s" instance to deduce the correct timestamp.', DateHeader::class));
+                }
+                // the documentation shows parameter names in camelCase but the expected JSON keys should be in snake_case
+                $payload['send_at'] = $header->getDateTime()->getTimestamp();
+            } elseif ($header instanceof TagHeader) {
```

Comment:
How about using guard clauses, instead of nested conditions, to improve readability?

https://github.com/symfony/symfony/blob/7.3/src/Symfony/Component/Mailer/Bridge/Postmark/Transport/PostmarkApiTransport.php#L114-L145

Reply:
Hi,thank you for your suggestion.
I didn't want to change too much of the code to facilitate the review process but I'll gladly update the successive conditions if necessary. Is it something you would like me to do in this PR ?

Reply:
It's just a suggestion, but since you added new conditions, this approach might make sense, as it's used in other integrations. Perhaps wait to hear other opinions on the matter.

Reply:
It's just a suggestion, but since you've added new conditions, this approach might make sense, as it's used in other integrations. I think it's worth waiting to hear and consider other opinions before making a final decision.